README.md

```diff
@@ -1,4 +1,129 @@
 # xbrl-parser
+
+[![GoDoc](https://godoc.org/github.com/polygon-io/xbrl-parser?status.svg)](https://godoc.org/github.com/polygon-io/xbrl-parser)
+
 A Go library to parse xbrl documents into their facts, contexts, and units.
 
-**Waring:** This library is under development and may change dramatically!
+This library is based around the [XBRL 2.1 spec](https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html).
+It implements support for parsing basic facts (not tuples of facts), contexts and units through the `xml.Unmarshaler` interface.
+ 
+See the package example in the godocs for how to unmarshal into the `XBRL` struct.
+
+This library supports basic validation that checks for malformed facts and broken references between facts and contexts/units (see `XBRL.Validate()`),
+but it does _not_ implement full semantic validation of XBRL documents.
+
+There are no abstractions added on-top of the XBRL data structure, which makes this library flexible and simple,
+but it also means you might have to read up a bit on how XBRL works to take full advantage of it.
+
+To give you a head start, here's some basics about XBRL:
+
+## What is XBRL?
+
+At a high level, XBRL is an XML spec intended to be the standard for digital business reporting.
+
+I can't give you a more precise definition because "digital business reporting" can mean many things, 
+and XBRL needs to work for all of them.
+
+XBRL is useless on its own, it relies on supplemental taxonomies that describe facts for specific use cases.
+For example the [US GAAP Financial Reporting Taxonomy](https://xbrl.us/xbrl-taxonomy/2021-us-gaap/) defines schemas for
+facts that relate to US GAAP financial reporting, which is used heavily in quarterly reports submitted to the SEC, among many other things.
+
+If you're here, I'm assuming you have a specific reason to want to parse XBRL documents, so I'm not going to go into more detail on this.
+
+Let's jump right into the main components of an XBRL document... 
+
+### Facts
+
+The core of any XBRL document: facts (or [items](https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html#_4.6) as the spec calls them)
+represent a single business measurement.
+Here's an example of a fact whose schema is defined in the [US GAAP Financial Reporting Taxonomy](https://xbrl.us/xbrl-taxonomy/2021-us-gaap/):
+
+```xml
+<us-gaap:EarningsPerShareBasic contextRef="c1" decimals="2" unitRef="u1">1.41</us-gaap:EarningsPerShareBasic>
+```
+
+A fact by itself is only a fragment of a useful piece of information. 
+In the above example we see that earnings per share is `1.41`,
+but we need more context around when this fact was true, and how it was measured.
```

Comment:
Should we include:
`in order to <something>.`
Where `<something>` is a more concrete activity that motivates why the reader wants more context?

Reply:
Added a bit about how there might be more than one of the same fact